### DISCUSSION TITLE
Using the STILT model for the monitoring of volcanic eruptions

### DISCUSSION THREAD

**Discussion by evagldn** [uataq/stilt#125] [Air Quality]
Hello, 

I am trying to adapt the available tutorials to model the volcanic eruptions and its pollutants dispersion in the atmosphere. I’ve studied the STILT model and worked the associated tutorials and I’m now trying to adapt them to simulate an emission of sulfur dioxide from the Etna and its dispersion. 
I would like to use STILT with other inputs to investigate the variation of SO2 in an area centered on the volcano which will be the only point of the grid with a non-zero emission value. As STILT is used to show the backward trajectories, I planned to replace in the script _run_stilt.r_ the file **receptors.rds** (the file available for the light-rail tutorial) with another raster file according to the same type (data.frame) that will be a (latitude, longitude) grid centered on the volcano. For the **emission.rds** file, I planned to recreate a RasterBrick file like the one in the tutorials, also with a grid (4 columns : lat, lon, SO2 emission value, time) centered on the volcano, but, as I previously said, with the whole "SO2 emission value" column equal to zero, except the line corresponding to the point (lat, lon) of the volcano. By doing that, I hope that the backward trajectories from each receptor point will illustrate the dispersion of the SO2 after being emitted by the volcano. 

So, I have a couple questions concerning these two input files, in order to recreate them and be able to restart STILT for my simulation. 
I tried to recreate a data.frame identical to the one of the **receptors.rds** file, but when I start the _run_stilt.r_ script, I only obtain output files in the "out/by-id" directory. The "out/footprints" remains empty, and I don't understand why since when I run the script with the file of the light-rail tutorial, everything looks fine. I noticed that the footprints directory is completed by the _simulation_step.r_ script, but I don't understand the difference between the files that disrupt STILT. 
Concerning the **emissions.rds** file, which is a pretty much bigger file with a lot of information, I also tried to recreate its structure but I can't test it for now, because I need to run _run_stilt.r_ correctly before I run the _tutorial.r_ script (that need a **emissions.rds** file), but in cases where it doesn't work, I would like to have, if this is possible, more information about its creation, and the origin of the data used to complete it. 

To finish with, I wonder about the ability of STILT to work on the « forward » mode, that can appear to be a choice maybe more intuitive in the context of my study around the volcanic eruptions.

If you have any idea concerning this adaptation of the STILT tutorials, I would be grateful for your help. I can of course precise anything that wasn't comprehensible enough in my explanations, or share my adapted scripts and inputs.

Thank you for your time

**Comment by tartanrunner25** [comment]
Hi evagldn,

For your first issue, I would make sure that the receptors object is being properly set in the code as this could potentially cause the simulation to fail and not generate output files, which is what you were seeing. I would compare this to the object crated by the tutorial you ran as a reference point. 

Running STILT works perfect fine in forward mode, here you would just set the time from a negative to a positive time. For example, if you run a backward simulation 24 hours, you would set n_hours as -24, but if you wanted to do this 24 hours forward in time you would just drop the negative.
